maps-app-ios/UI/Map View/MapViewController+AppPreferences.swift

```diff
@@ -29,7 +29,9 @@ extension MapViewController {
     
     func observeValueForPreferences(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?) {
         if let target = object as? AGSMapView, keyPath == navigationPropertyKey, let navigating = change?[.newKey] as? Bool, navigating == false {
-            mapsAppPrefs.viewpoint = target.currentViewpoint(with: .centerAndScale)
+            DispatchQueue.main.async {
```

Comment:
Does this need to be on the main queue? Would a serial background queue work as well?

Reply:
Thought I was getting errors in the console about having to run on the main thread. Let me revisit.

Reply:
And perhaps it's a lack of understanding what is being performed. As I understand it, what's happening here is by setting the viewPoint of the mapsAppPrefs is saving the view point JSON to UserDefaults. If there is some UI updates happening along the way that i'm missing then it would make sense to keep this on the main thread.